api/v1beta1/specialresource_types.go

```diff
@@ -54,94 +58,127 @@ type SpecialResourceArtifacts struct {
 	Claims []SpecialResourceClaims `json:"claims,omitempty"`
 }
 
-// SpecialResourceBuildArgs defines the observed state of SpecialResource
+// SpecialResourceBuildArgs is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceBuildArgs struct {
 	Name  string `json:"name"`
 	Value string `json:"value"`
 }
 
-// SpecialResourceConfiguration defines the observed state of SpecialResource
+// SpecialResourceConfiguration is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceConfiguration struct {
 	Name  string   `json:"name"`
 	Value []string `json:"value"`
 }
 
-// SpecialResourceGit defines the observed state of SpecialResource
+// SpecialResourceGit is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceGit struct {
 	Ref string `json:"ref"`
 	Uri string `json:"uri"`
 }
 
-// SpecialResourceSource defines the observed state of SpecialResource
+// SpecialResourceSource is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceSource struct {
 	Git SpecialResourceGit `json:"git,omitempty"`
 }
 
-// SpecialResourceDriverContainer defines the desired state of SpecialResource
+// SpecialResourceDriverContainer is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceDriverContainer struct {
 	// +kubebuilder:validation:Optional
 	Source SpecialResourceSource `json:"source,omitempty"`
+
 	// +kubebuilder:validation:Optional
 	Artifacts SpecialResourceArtifacts `json:"artifacts,omitempty"`
 }
 
-// SpecialResourceSpec defines the desired state of SpecialResource
+// SpecialResourceSpec describes the desired state of the resource, such as the recipe to be used and a selector
+// on which nodes it should be installed.
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
 type SpecialResourceSpec struct {
+	// Chart describes the Helm chart that needs to be installed.
 	// +kubebuilder:validation:Required
 	Chart helmerv1beta1.HelmChart `json:"chart"`
+
+	// Namespace describes in which namespace the chart will be installed.
 	// +kubebuilder:validation:Required
 	Namespace string `json:"namespace"`
+
+	// ForceUpgrade is not used.
 	// +kubebuilder:validation:Optional
 	ForceUpgrade bool `json:"forceUpgrade"`
+
+	// If Debug is true, additional debugging output will be written in the log file.
 	// +kubebuilder:validation:Optional
 	Debug bool `json:"debug"`
+
+	// Set are Helm hierarchical values for this chart installation.
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:pruning:PreserveUnknownFields
 	// +kubebuilder:validation:EmbeddedResource
 	Set unstructured.Unstructured `json:"set,omitempty"`
+
+	// DriverContainer is not used.
```

Comment:
It's not used in code, but I've seen it used just like a values from `Set` would be:
https://github.com/openshift-psap/special-resource-operator/blob/master/charts/example/simple-kmod-0.0.1/simple-kmod.yaml#L20
https://github.com/openshift-psap/special-resource-operator/blob/master/charts/example/simple-kmod-0.0.1/templates/0000-buildconfig.yaml#L28

Reply:
If it is not used in the code it is safe to remove even if not defined. Since we are using OpenAPI we take advantage of `additionalProperties` defaulting to null, which is equivalent to `true`, which means you can add whatever you want in an instance for this definition. It is going to be ignored later on.
Adding/removing mandatory files does need a new API version, though.
This depends on what our versioning strategy is in regards to optionals. I would change to v1beta2 and deprecate the old one, but that might be noisy with partners.

Reply:
Agreed. I'm looking into migrating the API to a new version with all these fields removed.